File: C++/Niveau 1/7 - Conditions avancées, opérateurs booléens/L'espion est démasqué !.cpp
#include <iostream>
using namespace std;

int main(){
    int nombrePersonnes;

    cin >> nombrePersonnes;

    for(int i = 1; i <= nombrePersonnes; i++){
        int taille, age, poids, cheval, cheveux;

        cin >> taille >> age >> poids >> cheval >> cheveux;

        int criteres = 0;

        if((taille >= 178) && (taille <= 182)){
            criteres++;
        }

        if(age >= 34){
            criteres++;
        }

        if(poids < 70){
            criteres++;
        }

        if(cheval == 0){
            criteres ++;
        }

        if(cheveux == 1){
            criteres++;
        }

        if(criteres == 0){
            cout << "Impossible" << endl;
        }else if(criteres == 5){
            cout << "Très probable" << endl;
        }else if((criteres >= 3) && (criteres <= 4)){
            cout << "Probable" << endl;
        }else{
            cout << "Peu probable" << endl;
        }

    }

}

File: Java/Niveau1/7- Conditions avancées, opérateurs booléens/L'espion est démasqué !.java
import algorea.Scanner;

class Main{
    static Scanner entrée = new Scanner(System.in);     

    public static void main(String[] args){
        int nombresPersonnes = entrée.nextInt();
        for(int i = 1; i <= nombresPersonnes; i++){
            int taille = entrée.nextInt();
            int age = entrée.nextInt();
            int poids = entrée.nextInt();
            int cheval = entrée.nextInt();
            int cheveuxBruns = entrée.nextInt();
            int critères = 0;
            if((taille <= 182) && (taille >= 178)){
                critères++;
            }
            if(age >= 34){
                critères++;            
            }
            if(poids < 70){
                critères++;
            }
            if(cheval == 0){
                critères++;
           }
           if(cheveuxBruns == 1){
                critères++;
            }

            if (critères == 5){
                print("Très probable");
            }else if( critères == 0){
                print("Impossible");
            }else if((critères == 3) || (critères == 4 )){
                print("Probable");
            }else{
                print("Peu probable");
            }

        }
        
    }

    private static void print(Object object){
        System.out.println(object);
    }
}
